### ISSUE TITLE
When a extreamly small metric value(close to 0) is ingested, storage save this value as `-Inf`.

### ISSUE THREAD

**Issue by dxtrzhang** [VictoriaMetrics/VictoriaMetrics#1114]
**Describe the bug**
When a extreamly small metric value(close to 0) is ingested, storage save this value as `-Inf`.

**To Reproduce**
Write these metric points:
```
foo{bar="baz"} 2.964393875E-314
```
Then query data via `export`, here is the output:
```json
{"metric":{"__name__":"foo","bar":"baz"},"values":[-Inf],"timestamps":[1614956358925]}
```

**Expected behavior**
Extreamly small value should respect `-precisionBits` and treated as `0` in the example above.  
Seems like some error happend when convert float to decimal at [positiveFloatToDecimalSlow](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/536d59914a09813b1339b73dbb5edccffd85fcb9/lib/decimal/decimal.go#L446) when ingest data.

**Version**
verified at `v1.39.4-cluster`, `v1.52.0-cluster` and `1.55.1-cluster`

**Comment by valyala** [comment]
@dxtrzhang , could you verify whether the bug is fixed in the commit 1187ee5e16939cb37a2dfaeb1af6dcab80c59367 ?

**Comment by dxtrzhang** [comment]
Newly add metric points seem work fine after commit [1187ee5](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/1187ee5e16939cb37a2dfaeb1af6dcab80c59367).

**Comment by valyala** [comment]
The bugfix has been included in VictoriaMetrics [v1.56.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
